runtime/lua/vim/treesitter/language.lua

```diff
@@ -14,6 +14,14 @@ function M.require_language(lang, path, silent)
     return true
   end
   if path == nil then
+    if not pcall(vim.fn.glob, lang, true) then
+      -- Catch any malformed lang values as nvim_get_runtime_file will hard-error
+      -- if a bad pattern is passed. e.g: '{'
+      if silent then
+        return false
+      end
+      error("no parser for '"..lang.."' language, see :help treesitter-parsers")
+    end
     local fname = 'parser/' .. lang .. '.*'
```

Comment:
probably we can just escape the value of lang with `fnameescape`, and maybe also escape `/`